CodeOfConduct.md

```diff
@@ -16,11 +16,46 @@ WebAssembly operates under the W3C's
 > vehicle for better identity of the organization.
 
 We hope that our community group act according to these guidelines, and that
-participants hold each other to these high standards. If you have any questions
-or are worried that the code isn't being followed, please contact the Community
-Group's chairs at `team-wasm-chairs@w3.org` (note: this list is also
-visible to W3C staff). For very serious concerns, the W3C has [procedures][]
-allowing you to access its ombuds directly and confidentially.
+participants hold each other to these high standards. 
+
+If you are worried that the code isn't being followed, or are witness to a violation but are
+uncomfortable speaking up,please contact the Community Group's chairs at
+`team-wasm-chairs@w3.org` (note: this list is also visible to W3C staff). 
+
+In cases where there is repeated disregard for the CEPC, it is possible to incur disciplinary
+action. The points below outline a series of steps that will be taken in case of violations
+to ensure a positive work environment in the WebAssembly CG. 
+
+ - **Informal warning**   
+   An informal or verbal warning can be issued in any relevant medium of communication callin 
+   out inappropriate behavior as a violation of the CEPC
+   (examples: responses to github issues or PRs, discord, email, meetings).
+
+ - **Corrective action and/or counseling**   
+   Informal meetings, or email discussions where problematic behavior is addressed,
+   and suggestions for constructive ways forward are offered. A mediation process can also be an
+   option if relevant to the situation and both parties agree. 
+
+ - **Formal warning**   
+   An explicit warning issued after repeated incidents. While members of the CG can call out
+   inappropriate behavior as needed, warnings will be issued by CG chairs, subgroup chairs, or
+   proposal champions depending on the nature of the violation.
+
+ - **Disciplinary meeting**  
+   An optional meeting to discuss constructive action or next steps.
```

Comment:
A meeting with whom?

Reply:
Currently this would be with the chairs and possibly with someone from the W3C if needed, in the future this could also take form of a committee. The details of that are not clear yet, but publishing this doc is an iterative step towards a more transparent process.

Reply:
At whose option is the meeting? Maybe optional means that it can be requested by the person being warned (such that they get the opportunity to speak for themselves before chairs can escalate to suspension)? But if the person declines or is unresponsive, then they can go ahead?

Reply:
I've updated this a little bit, PTAL. Also cc @linclark to make sure this looks right.